File: app/src/main/java/android/mb00/codehub/api/parser/DateParser.java
package android.mb00.codehub.api.parser;


public class DateParser {

    public static String parseStandard(String dateAndTime) {
        int endIndex = dateAndTime.indexOf("T");
        String date = dateAndTime.substring(0, endIndex);
        return date;
    }

    public static String parseEnglish(String dateAndTime) {
        String year = dateAndTime.substring(0, 4);
        String month = dateAndTime.substring(5, 7);
        String day = dateAndTime.substring(8, 10);

        if (day.charAt(0) == '0') {
            day = day.replace("0", "");
        }

        switch(month) {
            case "01":
                month = "Jan";
                break;
            case "02":
                month = "Feb";
                break;
            case "03":
                month = "Mar";
                break;
            case "04":
                month = "Apr";
                break;
            case "05":
                month = "May";
                break;
            case "06":
                month = "Jun";
                break;
            case "07":
                month = "Jul";
                break;
            case "08":
                month = "Aug";
                break;
            case "09":
                month = "Sep";
                break;
            case "10":
                month = "Oct";
                break;
            case "11":
                month = "Nov";
                break;
            case "12":
                month = "Dec";
                break;
        }

        String date = month + " " +  day + ", " + year;
        return date;
    }

}

File: app/src/main/java/mb00/android/codehub/logic/utils/DateParser.kt
package mb00.android.codehub.logic.utils

/**
 * Contains methods used to parse dates from the GitHub API
 */

object DateParser {

    fun parseStandard(dateAndTime: String?): String? {
        val endIndex = dateAndTime?.indexOf("T") ?: 0
        return dateAndTime?.substring(0, endIndex)
    }

    fun parseEnglish(dateAndTime: String?): String? {
        val year = dateAndTime?.substring(0, 4)
        var month = dateAndTime?.substring(5, 7)
        var day = dateAndTime?.substring(8, 10)

        if (day?.get(0) == '0') {
            day = day.replace("0", "")
        }

        when (month) {
            "01" -> month = "Jan"
            "02" -> month = "Feb"
            "03" -> month = "Mar"
            "04" -> month = "Apr"
            "05" -> month = "May"
            "06" -> month = "Jun"
            "07" -> month = "Jul"
            "08" -> month = "Aug"
            "09" -> month = "Sep"
            "10" -> month = "Oct"
            "11" -> month = "Nov"
            "12" -> month = "Dec"
        }

        return "$month $day, $year"
    }

}
